### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by ngoiyaeric** [google/comprehensive-rust#1393] [General]
Getting an error installing the course locally, here is the log. 
error: failed to compile `mdbook-exerciser v0.1.0 (/Users/karibu/comprehensive-rust/mdbook-exerciser)`, intermediate artifacts can be found at `/Users/karibu/comprehensive-rust/target`

Caused by:
  failed to select a version for the requirement `clap = "^4.0.29"`
  candidate versions found which didn't match: 3.2.25, 3.2.24, 3.2.23, ...
  location searched: crates.io index
  required by package `mdbook v0.4.25`
      ... which satisfies dependency `mdbook = "^0.4.25"` of package `mdbook-exerciser v0.1.0 (/Users/karibu/comprehensive-rust/mdbook-exerciser)`

**Comment by mgeisler** [comment]
Hi @ngoiyaeric, thanks for the report.

This is odd since we run this all the time when publishing the course. @qwandor, can you help debug this?

**Reply by qwandor** [reply]
It works fine for me on the tip of main without anything special. What exact commands did you use? What version of Rust are you using, and on what platform?

**Comment by mgeisler** [comment]
I tried

```
cargo update
cargo install --path mdbook-exerciser
```

and it worked fine right now. This was with Rust 1.73.0 from the stable channel.

Can you try running `cargo update` as well?

**Comment by ngoiyaeric** [comment]
Hi, I realized I hadn't installed rust, I thought it was within the steps for comprehensive rust. Perhaps it should be included in the preliminary steps for the course assuming people taking it have never coded rust before. anyway issue resolved.
